litestar/datastructures/secret_values.py

```diff
@@ -0,0 +1,94 @@
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from typing import Generic, TypeVar, Union
+
+__all__ = (
+    "SecretBytes",
+    "SecretString",
+    "SecretT",
+    "SecretValue",
+)
+
+# NOTE: Union instead of ` | ` is used for compatibility with Sphinx autodoc.
+#  - Sphinx autodoc doesn't handle string annotations for "bound" parameter in TypeVar.
+#  - We cannot use `|` without string annotations due to supporting < python 3.10.
+SecretT = TypeVar("SecretT", bound=Union[str, bytes])
+"""Type that represents a secret value of type ``str`` or ``bytes``."""
+
+
+class SecretValue(ABC, Generic[SecretT]):
+    """Represents a secret value that can be of type `str` or `bytes`."""
+
+    def __init__(self, secret_value: SecretT) -> None:
+        """Initializes a :class:`SecretValue` object with a secret value of type ``str`` or ``bytes``.
+
+        Args:
+            secret_value (str | bytes): The secret value to be encapsulated.
+        """
+
+        self._secret_value = secret_value
+
+    def get_secret(self) -> SecretT:
+        """Returns the actual secret value.
+
+        Returns:
+            str | bytes: The secret value.
+        """
+
+        return self._secret_value
+
+    @abstractmethod
+    def get_obscured(self) -> SecretT:
+        """Return the hidden representation of the secret value.
+
+        Raises:
+            NotImplementedError: Always raised to enforce implementation in subclasses.
+        """
+
+        raise NotImplementedError("Subclasses must implement _get_obscured_representation")
```

Comment:
Did you mean the following?

```suggestion
        raise NotImplementedError("Subclasses must implement get_obscured")
```

Reply:
thanks good catch!